File: go/questions/question_four.go
package questions

/*
		(define (compose f g)
  		(lambda (x)(f (g x))))
*/

func H(x int) int {
	square := func(x int) int { return x * x }
	inc := func(x int) int { return x + 1 }
	return square(inc(x))
}

File: java/src/main/java/QuestionFour.java
import java.util.function.Function;

public class QuestionFour {

	/*
		(define (compose f g)
  		(lambda (x)(f (g x))))
	 */

	public static int h(int x) {
		Function<Integer, Integer> inc = (integer -> integer + 1);
		Function<Integer, Integer> square = (integer -> integer * integer);
		return square.compose(inc).apply(x);
	}
}
